internal/domain/utils.go

```diff
@@ -0,0 +1,50 @@
+package domain
+
+import (
+	"math/rand"
+	"unicode"
+)
+
+type SignCount struct {
+	upperCaseCount int
+	lowerCaseCount int
+	digitCount     int
+}
+
+func charTypeCounts(password string) SignCount {
+	var upperCaseCount, lowerCaseCount, digitCount int
+
+	for _, char := range password {
+		switch {
+		case unicode.IsUpper(char):
+			upperCaseCount++
+		case unicode.IsLower(char):
+			lowerCaseCount++
+		case unicode.IsDigit(char):
+			digitCount++
+		}
+	}
+
+	return SignCount{
+		upperCaseCount: upperCaseCount,
+		lowerCaseCount: lowerCaseCount,
+		digitCount:     digitCount,
+	}
+}
+
+func generateRandomUppercase(numUpper int) string {
```

Comment:
What is numUpper?

Reply:
The parameter specifies how many random uppercase letters the function should return. I see that the name is not understandable, I changed it to `uppercaseCount`